resources/build/builder.inc.sh

```diff
@@ -0,0 +1,688 @@
+#!/usr/bin/env bash
+#
+# This script contains utilities for builder_script calls
+#
+# * builder_ functions and variables are defined here.
+# * REPO_ROOT defines the top level of this repository
+# * THIS_SCRIPT_PATH defines the full path of the running script
+# * THIS_SCRIPT_NAME defines the basename of the running script
+# * THIS_SCRIPT_IDENTIFIER defines the repo-relative path of the running script
+# * _builder_ functions and variables are internal use only for this script 
+#   and subject to change
+#
+
+# _builder_init is called internally at the bottom of this file after we have
+# all function declarations in place.
+function _builder_init() {
+  _builder_findRepoRoot
+  _builder_setBuildScriptIdentifiers
+
+  if [[ -n "$TERM" ]] && [[ "$TERM" != "dumb" ]] && [[ "$TERM" != "unknown" ]]; then
+    builder_use_color true
+  else
+    builder_use_color false
+  fi
+}
+
+function _builder_findRepoRoot() {
+    # See https://stackoverflow.com/questions/59895/how-to-get-the-source-directory-of-a-bash-script-from-within-the-script-itself
+    # None of the answers are 100% correct for cross-platform
+    # On macOS, requires coreutils (`brew install coreutils`)
+    local SCRIPT=$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")
+    REPO_ROOT=$(dirname $(dirname $(dirname "$SCRIPT")))
```

Comment:
Once we move this file up a folder, we'll need to remove one of these dirname calls:
```suggestion
    REPO_ROOT=$(dirname $(dirname "$SCRIPT"))
```